src/input_common.rs

```diff
@@ -484,7 +484,7 @@ impl Drop for TerminalProtocols {
         let sequences = concat!(
             "\x1b[?2004l",
             "\x1b[>4;0m",
-            "\x1b[<1u", // Konsole breaks unless we pass an explicit number of entries to pop.
+            "\x1b[<9u", // Pop a few number of entries as a best effort to disable kitty progressive enhancement.
```

Comment:
we should never need to pop more than one set of flags because we never push more than one set at a time.
It seems like the motivation for changing this is to work around buggy child processes, which I'm not sure we should do. The problems are that we'd mask bugs in child processes, and we'd break nesting (by popping flags set by a parent process).

I'm not sure what's your motivating example but to give a concrete one, if I run `fish` and then run Kakoune from fish, then a crashing Kakoune will leak progressive enhancement flags. It seems appropriate to fix the leak in Kakoune (fish does that when it panics, which covers most cases).

Reply:
>  work around buggy child processes,

The goal is to address the issue of crashing child processes. This can be easily reproduced by running a child fish shell and then using `kill -9` to terminate it. After doing this, any newly created child processes will always have the kitty keyboard protocol enabled, regardless of whether they support it or not.

Reply:
any particular reason why your child process crashes? Usually it should be possible to clean up in a crash handler.

I think the expectation that a crashing child process not leak terminal state makes sense.
However we should not implement it by violating the protocol. Instead we could extend the [protocol](https://sw.kovidgoyal.net/kitty/keyboard-protocol/#progressive-enhancement) to maybe allow both getting the stack size and truncating the stack later

Reply:
> any particular reason why your child process crashes? 

"Crash" is a general term in this context. A more specific example is the loss of connection to a remote shell.

Reply:
right, when doing `ssh -> fish`, then a disconnect will leave the terminal in a bad state.
We could extend the protocol as described above and use that in SSH, to restore flags upon exit (assuming that flags are never leaked intentionally). Then it will work also if the local shell is not fish.
Until then, we could stop pushing protocols if `$SSH_TTY` is set but `$TMUX` (or equivalent) is not? Is there any other relevant scenario?

Reply:
Yeah actually I think the benefits of the stack are somewhat theoretical, so if we don't find a better solution, we should abandon the stack and set/clear flags directly.

I think the best case is if we can change SSH to clean up. I've openend https://github.com/kovidgoyal/kitty/issues/7603 to summarize my thoughts for a different audience.

As for gauging the importance of this issue. I think if `ctrl-p` stops working in bash that's pretty bad. The connection drop scenario is an everyday one, so we should fix it. Any crashes are not so important I think; those cases should be rare that it's reasonable to run `reset` or restart the terminal

Reply:
I respectfully disagree, if we assume crashes don't happen then the bulk of the reasoning for managing terminal state ourselves goes away altogether.

Reply:
> if we assume crashes don't happen

we don't have numbers but the scenario of "SSH disconnect, then run a program that doesn't support CSI u" sounds like it could happen once in a while to the unsuspecting user.
At the same time, I'm still trying to find a tangible benefit for using the flag stack at all; it seems better to set/clear the flags directly, to fix the above scenario.